mlir/lib/IR/AsmPrinter.cpp

```diff
@@ -780,9 +783,10 @@ class DummyAliasOperationPrinter : private OpAsmPrinter {
   /// Print the given set of attributes with names not included within
   /// 'elidedAttrs'.
   void printOptionalAttrDict(ArrayRef<NamedAttribute> attrs,
-                             ArrayRef<StringRef> elidedAttrs = {}) override {
-    if (attrs.empty())
-      return;
+                             ArrayRef<StringRef> elidedAttrs = {},
+                             function_ref<LogicalResult(NamedAttribute)>
+                                 printNamedAttrFn = nullptr) override {
+    (void)printNamedAttrFn;
```

Comment:
Why not just use `printNamedAttrFn` when not NULL?

Reply:
This is the DummyAliasOperationPrinter. It does not actually print anything, just tracks what would be printed (e.g. which attributes). A custom print style would not change that.